src/installer/pkg/sfx/installers/host.wxs

```diff
@@ -20,38 +20,48 @@
         </File>
       </Component>
         
-      <Component Id="cmpSharedHostVersionRegistry" Directory="DOTNETHOME" Guid="*">
+      <Component Id="cmpSharedHostVersionRegistry" Directory="DOTNETHOME">
         <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sharedhost">
           <RegistryValue KeyPath="yes" Action="write" Name="Version" Type="string" Value="$(var.NugetVersion)"/>
         </RegistryKey>
       </Component>
 
-      <Component Id="cmpInstallLocation" Directory="TARGETDIR" Guid="*" Win64="no">
+      <Component Id="cmpInstallLocation" Directory="TARGETDIR" Bitness="always32">
         <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)">
           <RegistryValue Action="write" Name="InstallLocation" Type="string" Value="[DOTNETHOME]" KeyPath="yes"/>
         </RegistryKey>
       </Component>
 
-      <Component Id="cmpPath" Directory="DOTNETHOME" Guid="*">
-        <?if $(var.Platform)~=x64 ?>
+      <?if $(var.Platform)~=x64 ?>
         <!-- For x64 installer, only add the sharedhost key when actually on native architecture. -->
-        <Condition>NOT NON_NATIVE_ARCHITECTURE</Condition>
-        <?elseif $(var.Platform)~=x86 ?>
+        <Component Id="cmpPath" Directory="DOTNETHOME" Condition="NOT NON_NATIVE_ARCHITECTURE">
+          <!-- A stable keypath with the right SxS characteristics for our PATH entry-->
+          <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sharedhost">
+            <RegistryValue KeyPath="yes" Action="write" Name="Path" Type="string" Value="[DOTNETHOME]"/>
+          </RegistryKey>
+        </Component>
+      <?elseif $(var.Platform)~=x86 ?>
         <!-- For x86 installer, only add the key when not on 64-bit platform. -->
-        <Condition>NOT VersionNT64</Condition>
-        <?endif?>
-
-        <!-- A stable keypath with the right SxS characteristics for our PATH entry-->
-        <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sharedhost">
-          <RegistryValue KeyPath="yes" Action="write" Name="Path" Type="string" Value="[DOTNETHOME]"/>
-        </RegistryKey>
-      </Component>
+        <Component Id="cmpPath" Directory="DOTNETHOME" Condition="NOT VersionNT64">
+          <!-- A stable keypath with the right SxS characteristics for our PATH entry-->
+          <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sharedhost">
+            <RegistryValue KeyPath="yes" Action="write" Name="Path" Type="string" Value="[DOTNETHOME]"/>
+          </RegistryKey>
+        </Component>
+      <?else?>
+        <Component Id="cmpPath" Directory="DOTNETHOME">
+          <!-- A stable keypath with the right SxS characteristics for our PATH entry-->
+          <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sharedhost">
```

Comment:
nit: you can replace all the `$(var.` with just `$(`

Reply:
Left this as is since we use `var` in most other places too (arcade). Might be worth a cleanup pass in net11.